pkg/profilecreator/profilecreator.go

```diff
@@ -266,31 +295,38 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 		} else {
 			max = reservedCPUCount
 		}
-		//sorting by comparing the first logical core of processor cores
-		sort.Slice(node.Cores, func(i, j int) bool {
-			//sorting the logical cores a processor
-			sort.Slice(node.Cores[i].LogicalProcessors, func(x, y int) bool {
-				return node.Cores[i].LogicalProcessors[x] < node.Cores[i].LogicalProcessors[y]
-			})
-			sort.Slice(node.Cores[j].LogicalProcessors, func(x, y int) bool {
-				return node.Cores[j].LogicalProcessors[x] < node.Cores[j].LogicalProcessors[y]
-			})
-			return node.Cores[i].LogicalProcessors[0] < node.Cores[j].LogicalProcessors[0]
-		})
+		if max%2 != 0 && htEnabled {
```

Comment:
the more I read about this allocation loop, the more I wonder if it would'nt be much more readable to just have two completely separate helper functions, one per allocation strategy (splitReservedCPUsAcrossNUMA=false/true -> two strategies), each one with his clear and nice allocation loop.

Can be done later, however.

Reply:
Sure, the allocation logic can certainly be made more readable! I will add a TODO along with the suggested approach and do the refactoring in a separate PR.